src/component/chatting/inputChat.tsx

```diff
@@ -0,0 +1,106 @@
+import { useState } from 'react';
+import styled from 'styled-components';
+import { useRecoilState, useRecoilValue } from 'recoil';
+import { chatList, selectedUser } from '../../atom/atom';
+import { ChatInfo, ChattingRoom } from '../../interface/interface';
+
+const InputChat = ({ chattingRoomId }: ChattingRoom) => {
+  const [text, setText] = useState<string>('');
+  const [newChatList, setNewChatList] = useRecoilState(chatList);
+  const [realChat, setRealChat] = useState(newChatList[chattingRoomId].message);
+  const selectedId = useRecoilValue(selectedUser);
+
+  let updateChat;
+
+  const submitText = (e?: React.FormEvent<HTMLFormElement>) => {
+    e?.preventDefault();
+    if (text.trim()) {
+      let newChat: ChatInfo = {
+        userNum: selectedId,
+        addText: text,
+        date: String(new Date()),
+      };
+
+      updateChat = realChat.concat(newChat);
+      setRealChat(updateChat);
+
+      const finalChatList = {
+        ...newChatList[chattingRoomId],
+        message: updateChat,
+      };
```

Comment:
realChat 라는 변수가 finalChatList를 업데이트하기 위한 updateChat의 갱신에만 쓰이는 거라면,
realChat를 따로 정의해주지 않고 newChatList을 통해 리스트를 업데이트 해줄 수 있을 것 같아요!
```suggestion
      // updateChat = realChat.concat(newChat);
      // setRealChat(updateChat);

      const finalChatList = {
        ...newChatList[chattingRoomId],
        message: [...newChatList[chattingRoomId].message, newChat],
      };
```

Reply:
오... 확실히 이 방법이 더 효율적인 것 같아요!!!